timescaledb/tutorials/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,347 @@
+{'event': 'price', 'symbol': 'ETH/USD', 'currency_base': 'Ethereum', 'currency_quote': 'US Dollar', 'exchange': 'Huobi', 'type': 'Digital Currency', 'timestamp': 1652438899, 'price': 2089.07, 'bid': 2089.02, 'ask': 2089.03, 'day_volume': 193818}
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438900, 'price': 30346.0, 'bid': 30346.0, 'ask': 30346.0, 'day_volume': 49167}
+```
+
+Each price event gives you multiple data points about the given trading pair.
+For example, the name of the exchange, current price, etc... You will
+occasionally see `heartbeat` events in the response - these events signal
+the health of the connection over time.
+
+At this point the websocket connection works and data keeps flowing. You need
+to implement the `on_event` function so data gets ingested into TimescaleDB.
+
+## Ingesting websocket data into TimescaleDB
+
+Now that the websocket connection is set up, implement the `on_event` function
+so data gets inserted into the database. When ingestin data into a
+transactional database, generally speaking, ingesting a thousand records in
+one transaction will be faster than ingesting them in a thousand transactions.
+
+TimescaleDB, being a transactional database, creates a new transaction
+whenever you try to insert a new record. Because of this behavior, it's suggested to
+insert data records in a batch - multiple records in one transaction.
+Using this approach you can achieve higher ingest rates than if you were to
+insert data record-by-record.
+
+When you insert data in batches, as opposed to one record at a time, the
+database spends less time managing the connection, transactions, etc... and
+ingestion becomes faster overall. 
+
+### Batching in memory
+
+A common practice to implement batching is to store new records in memory
+first, then after the batch reaches a certain size, insert all the records
+from memory into the database in one transaction. The perfect batch size isn't
+universal, but you can experiment with different batch sizes
+(eg.: 100, 1000, 10000 etc...) and see which one fits your use case better.
+Using batching is a fairly common pattern when ingesting data into TimescaleDB
+from Kafka, Kinesis, or websocket connections.
+
+Now that you know you need to queue up your data before ingesting, let's see
+how to implement a batching solution in Python with Psycopg2.
+
+### Implement batching with Psycopg2
+
+Remember to implement the ingestion logic within the `on_event` function that
+you then pass over to the websocket object.
+
+Let's break down what this function needs to do:
+
+1. Check if the item is a data item (and not some websocket metadata)
+2. Adjust the data so it fits the database schema (data types, order of columns)
+3. Add it to the in-memory batch (a list in Python)
+4. If the batch reaches a certain size, insert the data and reset/empty the list
```

Comment:
```suggestion
1. Check if the item is a data item, and not some websocket metadata.
1. Adjust the data so that it fits the database schema, including the data types, and order of columns.
1. Add it to the in-memory batch, which is a list in Python.
1. If the batch reaches a certain size, insert the data and reset or empty the list.
```